fs/exists_test.ts

```diff
@@ -6,128 +6,445 @@ import { exists, existsSync } from "./exists.ts";
+  } finally {
+    if (Deno.build.os !== "windows") {
+      await Deno.chmod(tempDirPath, 0o755);
+    }
+    await Deno.remove(tempDirPath, { recursive: true });
+  }
 });
 
-Deno.test("[fs] existsLink", async function () {
-  // TODO(axetroy): generate link file use Deno api instead of set a link file
-  // in repository
-  assertEquals(await exists(path.join(testdataDir, "0-link")), true);
+Deno.test("[fs] existsDirSync", function () {
+  const tempDirPath: string = Deno.makeTempDirSync();
+  try {
+    assertEquals(existsSync(tempDirPath), true);
+    assertEquals(existsSync(tempDirPath, {}), true);
+    assertEquals(
+      existsSync(tempDirPath, {
+        isDirectory: true,
+      }),
+      true,
+    );
+    assertEquals(
+      existsSync(tempDirPath, {
+        isFile: true,
+      }),
+      false,
+    );
+    if (Deno.build.os !== "windows") {
+      // TODO(martin-braun): include mode check for Windows tests when chmod is ported to NT
+      Deno.chmodSync(tempDirPath, 0o000);
+      assertEquals(
+        existsSync(tempDirPath, {
+          isReadable: true,
+        }),
+        false,
+      );
+    }
+  } catch (error) {
+    throw error;
+  } finally {
+    if (Deno.build.os !== "windows") {
+      Deno.chmodSync(tempDirPath, 0o755);
+    }
+    Deno.removeSync(tempDirPath, { recursive: true });
+  }
 });
 
-interface Scenes {
-  read: boolean; // --allow-read
-  async: boolean;
-  output: string;
-  file: string; // target file to run
+Deno.test("[fs] existsDirLinkSync", function () {
+  const tempDirPath: string = Deno.makeTempDirSync();
+  const tempLinkDirPath: string = path.join(tempDirPath, "temp-link");
+  try {
+    Deno.symlinkSync(tempDirPath, tempLinkDirPath);
+    assertEquals(existsSync(tempLinkDirPath), true);
+    assertEquals(existsSync(tempLinkDirPath, {}), true);
+    assertEquals(
+      existsSync(tempLinkDirPath, {
+        isDirectory: true,
+      }),
+      true,
+    );
+    assertEquals(
+      existsSync(tempLinkDirPath, {
+        isFile: true,
+      }),
+      false,
+    );
+    if (Deno.build.os !== "windows") {
+      // TODO(martin-braun): include mode check for Windows tests when chmod is ported to NT
+      Deno.chmodSync(tempDirPath, 0o000);
+      assertEquals(
+        existsSync(tempLinkDirPath, {
+          isReadable: true,
+        }),
+        false,
+      );
+      // TODO(martin-braun): test unreadable link when Rust's nix::sys::stat::fchmodat has been implemented
+    }
+  } catch (error) {
+    throw error;
+  } finally {
+    if (Deno.build.os !== "windows") {
+      Deno.chmodSync(tempDirPath, 0o755);
+    }
+    Deno.removeSync(tempDirPath, { recursive: true });
+  }
+});
+
+/**
+ * Scenes control additional permission tests by spawning new Deno processes with and without --allow-read flag.
+ */
+interface Scene {
+  read: boolean; // true to test with --allow-read
+  sync: boolean; // true to test sync
+  fictional: boolean; // true to test on non existing file
```

Comment:
`fictional` sounds confusing to me. How about `existing` (with reversed boolean value)?